### PR TITLE
docs: Update contributing guide pages

### PR DESCRIPTION
**Pull request by sayboras**
### Description

I am refreshing [contributing guide](https://docs.cilium.io/en/latest/contributing/development/), and realise that 
there are some simple discrepancies, hence just make some small changes.

Please refer to individual commits for details.

```release-note
docs: Update contributing guide pages
```

### CODE REVIEW

File: Documentation/contributing/development/contributing_guide.rst [hunk 1]
```diff
@@ -215,8 +215,10 @@ Getting a pull request merged
    ``/test`` as described in :ref:`trigger_phrases`. If you are a core team
    member, you may trigger the CI run yourself.
 
-   #. Hound: basic ``golang/lint`` static code analyzer. You need to make the
-      puppy happy.
+   #. Basic static code analyzer by Github Action and Travis CI. If a job is
+      failed, please refer to build log for required action (e.g. Please run
```
Comment (aditighag) [inline Documentation/contributing/development/contributing_guide.rst]: Golang linter suggestions are added in-line on PRs.

**Review by aditighag — APPROVED** [review]
Thanks! Posted a minor suggestion around the golang/lint note.

**Review by glibsm — APPROVED** [review]
assuming resolution of Aditi's comment

File: Documentation/contributing/development/contributing_guide.rst (hunk 1 above)
Comment (sayboras) [reply Documentation/contributing/development/contributing_guide.rst]: thanks, I didn't know about this feature as well, I have local post-push git hook to run `make lint precheck` :100:

File: Documentation/contributing/development/contributing_guide.rst [hunk 2]
```diff
@@ -215,8 +215,11 @@ Getting a pull request merged
    ``/test`` as described in :ref:`trigger_phrases`. If you are a core team
    member, you may trigger the CI run yourself.
 
-   #. Hound: basic ``golang/lint`` static code analyzer. You need to make the
-      puppy happy.
+   #. Basic static code analyzer by Github Action and Travis CI. Golang linter
+      suggestions are added in-line on PRs. For other failed jobs, please refer
+      to build log for required action (e.g. Please run 'go mod tidy && go mod
+      vendor' and submit your changes, etc).
```
Comment (qmonnet) [inline Documentation/contributing/development/contributing_guide.rst]:
```suggestion
      to build log for required action (e.g. Please run ``go mod tidy && go mod
      vendor`` and submit your changes, etc).
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
One nit on formatting but looks good, thanks for the dusting!

File: Documentation/contributing/development/contributing_guide.rst (hunk 2 above)
Comment (sayboras) [reply Documentation/contributing/development/contributing_guide.rst]: thanks :100:

**Review by qmonnet — APPROVED** [review]

**Pull request merged**